### PR TITLE
switch to native gnupg and fix GPG_PATH too long issue

### PR DESCRIPTION
**Pull request by yunguowei**

### CODE REVIEW

File: meta/classes/sign_rpm.bbclass [hunk 1]
```diff
@@ -19,7 +19,7 @@ RPM_SIGN_PACKAGES='1'
 python () {
     # Check RPM_GPG_NAME configuration
     if not d.getVar('RPM_GPG_NAME', True):
-        raise_sanity_error("You need to define RPM_GPG_NAME in the config")
+        raise_sanity_error("You need to define RPM_GPG_NAME in the config", d)
```
Comment (jiazhang0) [inline meta/classes/sign_rpm.bbclass]: Change the error message to " RPM_GPG_NAME is undefined in the config".

File: meta/classes/sign_rpm.bbclass [hunk 2]
```diff
@@ -29,6 +29,13 @@ python () {
     if not d.getVar('GPG_PATH', True):
         d.setVar('GPG_PATH', d.getVar('DEPLOY_DIR_IMAGE', True) + '/.gnupg')
 
+    # if the length of gpg home directory is over 94, gpg2 will return error
+    # during key importing and rpm signing. In this case, we return sanity
+    # error and let user reconfigure proper GPG_PATH
+    gpg_path = d.getVar('GPG_PATH', True).strip()
+    if len(gpg_path) > 94:
+        raise_sanity_error("GPG_PATH: %s is too long, and please limit it within 94 characters." % gpg_path, d)
```
Comment (jiazhang0) [inline meta/classes/sign_rpm.bbclass]: Change the message to "GPG_PATH %s ..."

Comment (jiazhang0) [inline meta/classes/sign_rpm.bbclass]: If gpg version 1 doesn't have such a 94-byte limit, we could switch to use it as the native app.

**Comment by yunguowei** [comment]
I will close it an send a new one

**Pull request closed without merging**